python/ray/experimental/sgd/pytorch/pytorch_runner.py

```diff
@@ -0,0 +1,163 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+import os
+import torch
+import torch.distributed as dist
+import torch.utils.data
+
+import ray
+
+from ray.experimental.sgd.pytorch import utils
+
+logger = logging.getLogger(__name__)
+
+
+class PyTorchRunner(object):
+    """Manages a distributed PyTorch model replica.
+
+    Args:
+        model_creator (dict -> torch.nn.Module): creates the model using the
+            config.
+        data_creator (dict -> Dataset, Dataset): creates the training and
+            validation data sets using the config.
+        optimizer_creator (model, dict -> loss, optimizer): creates the loss
+            and optimizer using the config.
+        config (dict): configuration passed to 'model_creator', 'data_creator',
+            and 'optimizer_creator'.
+        batch_size (int): batch size used for SGD.
+        backend (string): backend used for distributed SGD. "gloo" or "nccl".
+    """
+
+    def __init__(self,
+                 model_creator,
+                 data_creator,
+                 optimizer_creator,
+                 config=None,
+                 batch_size=16,
+                 backend="gloo"):
+
+        self.model_creator = model_creator
+        self.data_creator = data_creator
+        self.optimizer_creator = optimizer_creator
+        self.config = {} if config is None else config
+        self.batch_size = batch_size
+        self.backend = backend
+        self.verbose = True
+
+        self.local_rank = None
+        if os.environ.get("CUDA_VISIBLE_DEVICES", None):
+            # TODO: might break if multiple GPUs requested
```

Comment:
Will definitely break if multiple GPUs are requested. Maybe include an assertion here that there aren't more than 1 GPUs and prohibit more than 1 GPU in the resource dict in the main `DistributedPyTorch` constructor?